commandlineparser/src/Parser.cpp

```diff
@@ -55,16 +55,39 @@ bool cmdParser::Parser::Parse(int argc, char*argv[])
 	return true;
 }
 
-void cmdParser::Parser::default_help() const
+std::vector<std::string> cmdParser::Parser::help_qualifier_keys_finder()
+{
+	std::vector<std::string> help_keys;
+	for (auto x : command_list)
+	{
+		if ((x.first.at(0) == '-') && (x.first.at(1) == '-'))
+		{
+			help_keys.push_back(x.first);
+		}
+		else if (x.second->get_option_long_command().empty())
+		{
+			help_keys.push_back(x.first);
+		}
+	}
+	return help_keys;
+
+}
+
+void cmdParser::Parser::default_help() 
 {
 	std::cout << "\n **************WELCOME TO OPTIONS OF THE LIBRARY *********************" << std::endl;
 	std::cout << "[FORMAT]" << std::endl;
 	std::cout << "-------------------------------------------------------------------------------------" << std::endl;
 	std::cout << "Short_Command" << "\t" << "Long_Command" << "\t" << "Short_Description" << "\t\t" << "Long_Description" << std::endl;
 	std::cout << "-------------------------------------------------------------------------------------" << std::endl;
-	for (auto it = help_command_list.begin(); it != help_command_list.end(); ++it)
-	{
-		std::cout << "\n" << it->second->get_option_short_command() << "\t\t" << it->second->get_option_long_command() << "\t\t" << it->second->get_option_short_description() << "\t\t" << it->second->get_option_long_description() << std::endl << std::endl;
-	}
-
+	
+	std::vector<std::string>  keys = help_qualifier_keys_finder();
+	
+	auto print = [&](const std::string target_key) 
+	{     
+		auto itr=command_list.find(target_key);
+		std::cout<<"\n"<< itr->second->get_option_short_command()<<"\t\t"<< itr->second->get_option_long_command()<<"\t\t" << itr->second->get_option_short_description()<<"\t\t"<<itr->second->get_option_long_description()<<std::endl;
```

Comment:
auto itr=command_list.find(target_key);

target_key->get_option_short_description() ......

Reply:
instead of,
 auto itr=command_list[target_key];
following has been done to revive the const-ness of  default_help().
auto itr=command_list.at(target_key);